File: leetcode/mediantwoarrays/mediantwoarrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int size1, size2, size3;
        int index1, index2;
        
        size1 = nums1.size();
        size2 = nums2.size();
        size3 = size1 + size2;
        index1 = index2 = size3 / 2;
        if ((size3 % 2) == 0) {
            index2 = index1 - 1;
        }
        
        int i = 0;
        int j = 0;
        int mean1 = 0;
        int mean2 = 0;
        int count = 0;
        while (count <= index1) {
            if (i == size1) {
                mean1 = nums2[j];
                j++;
            } else if (j == size2) {
                mean1 = nums1[i];
                i++;
            } else if (nums1[i] < nums2[j]) {
                mean1 = nums1[i];
                i++;
            } else {
                mean1 = nums2[j];
                j++;
            }
            
            if (count == index2) {
                mean2 = mean1;
            }
            count++;
        }
        
        return (double(mean1) + double(mean2))/2; 
    }
};

File: leetcode/mediantwoarrays/mediantwoarrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        """
        nums3 = list()
        nums3.extend(nums1)
        nums3.extend(nums2)
        nums3.sort()
        length = len(nums3)
        if (length % 2) == 0:
            mean = nums3[length / 2] + nums3[(length / 2) - 1]
            return float(mean) / 2
        else:
            return nums3[length / 2]
        """
        len1 = len(nums1)
        len2 = len(nums2)
        len3 = len1 + len2
        even = False
        mean_index1 = len3 / 2 + 1
        if (len3 % 2) == 0:
            even = True
            mean_index2 = (len3 / 2)

        mean_value1 = 0
        mean_value2 = 0
        i = 0
        j = 0
        count = 0
        while (count < mean_index1):
            if i == len1:
                val = nums2[j]
                j = j + 1
            elif j == len2:
                val = nums1[i]
                i = i + 1
            elif nums1[i] <= nums2[j]:
                val = nums1[i]
                i = i + 1
            else:
                val = nums2[j]
                j = j + 1
            count = count + 1
            
            if even and count == mean_index2:
                mean_value2 = val
            elif count == mean_index1:
                mean_value1 = val
                
        if even:
            return float(mean_value1 + mean_value2) / 2
        return mean_value1
